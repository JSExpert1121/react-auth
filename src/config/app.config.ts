export const KAKAOMAP_API_KEY = '44a509cf4bf161e250c940454df5977c';
export const SUPPORTED_COUNTRIES = ['kr', 'jp', 'cn', 'hk', 'sg', 'tw', 'th', 'my', 'au', 'us', 'gb', 'de', 'fr', 'kz', 'ae', 'in'];
// export const COUNTRY_CODES = ['82', '81', '86', '852', '65', '886', '66', '60', '61', '1', '44', '49', '49', '33', '7', '971', '91'];
