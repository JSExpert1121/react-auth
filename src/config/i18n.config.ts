/**
 * @name i18n.config
 * @description config file for i18n. currently support ko and en
 * @version 1.0
 * @todo add locales
 * @author Darko
 */
const enMessages = require('translations/en.json');
const koMessages = require('translations/ko.json');

export const DEFAULT_LOCALE = 'en';

export const appLocales = [
    { code: 'en', name: 'English' },
    { code: 'ko', name: 'Korean' }
];

// auxilliary function to generate messages for each locales
export const formatMessages: (locale: string, messages: Record<string, string>) => Record<string, string> = (locale, messages) => {
    const defaultFormattedMessages =
        locale !== DEFAULT_LOCALE
            ? formatMessages(DEFAULT_LOCALE, enMessages)
            : {};
    const flattenFormattedMessages = (formattedMessages: Record<string, string>, key: string) => {
        const formattedMessage =
            !messages[key] && locale !== DEFAULT_LOCALE
                ? defaultFormattedMessages[key]
                : messages[key];
        return Object.assign(formattedMessages, { [key]: formattedMessage });
    };
    return Object.keys(messages).reduce(flattenFormattedMessages, {});
};

export const translationMessages = {
    en: formatMessages('en', enMessages),
    ko: formatMessages('ko', koMessages),
};
