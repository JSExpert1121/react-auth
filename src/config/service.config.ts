export const API_GATEWAY_ADDRESS = 'https://apig.t.moria.hitit.xyz:8765';
export const OAUTH_CLIENT_ID = 'web';
export const OAUTH_CLIENT_SECRET = 'hititweb42!';