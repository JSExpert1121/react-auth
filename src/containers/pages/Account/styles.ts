import { Theme, createStyles } from '@material-ui/core/styles';
import Colors from 'assets/colors';

export const styleFn = (theme: Theme) => createStyles({
    root: {
        width: '100%',
        padding: theme.spacing(3),
        maxWidth: 512,
        minWidth: 480,
    },
    form: {
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        width: '100%',
        textAlign: 'center'
    },
    logo: {
        padding: theme.spacing(2, 0)
    },
    title: {
        fontSize: '1.5rem',
        color: 'rgba(0, 0, 0, 0.8)',
        padding: theme.spacing(2)
    },
    text: {
        fontSize: '0.875rem',
        color: 'rgba(0, 0, 0, 0.7)',
        padding: theme.spacing(2)
    },
    success: {
        color: 'rgba(0, 0, 0, 0.9)',
        padding: theme.spacing(1)
    },
    error: {
        fontSize: '0.875rem',
        color: Colors.error,
        padding: theme.spacing(0, 0, 0, 2)
    },
    busy: {
        width: '100%',
    },
    submit: {
        margin: theme.spacing(2, 0),
        height: 'calc(100% - 12px)'
    },
    bottomMenu: {
        width: '100%',
        display: 'flex',
        padding: theme.spacing(2, 0)
    }
});