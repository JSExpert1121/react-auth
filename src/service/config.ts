import { oauthClient } from './base';

export const setLanguage = (lang: string) => {
    oauthClient.setLanguage(lang);
}
