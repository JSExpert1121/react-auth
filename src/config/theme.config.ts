import { createMuiTheme } from '@material-ui/core/styles';

const theme = createMuiTheme({
	palette: {
		primary: {
			main: '#3f51b5',
			dark: '#303f9f',
		},
		secondary: {
			light: '#0033EE',
			main: '#1100BB',
			contrastText: '#000000',
		},
		background: {
			default: '#FFF'
		}
	},
	typography: {
		fontFamily: "\"Noto Sans\", \"Noto Sans KR\", sans-serif"
	},
	overrides: {
		MuiAppBar: {
			colorPrimary: {
				color: '#111',
				backgroundColor: 'white'
			}
		},
		MuiContainer: {
			root: {
				padding: 0
			}
		},
		MuiBadge: {
			colorSecondary: {
				color: 'white'
			}
		},
		MuiExpansionPanel: {
			root: {
				margin: '8px 0',
				'&:before': {
					height: 0
				}
			}
		},
		MuiExpansionPanelSummary: {
			root: {
				padding: 0,
				'&:before': {
					height: 0
				}
			},
			content: {
				margin: 0,
				flexGrow: 0,
				'&.Mui-expanded': {
					margin: 0
				}
			}
		},
		MuiExpansionPanelDetails: {
			root: {
				padding: 8
			}
		}
	}
});

export default theme;
