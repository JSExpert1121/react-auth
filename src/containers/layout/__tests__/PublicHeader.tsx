import React from 'react';
import { MemoryRouter, withRouter } from 'react-router-dom';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16/build';
import { createMount } from '@material-ui/core/test-utils';
import { ThemeProvider } from '@material-ui/styles';
import { IntlProvider, injectIntl } from "react-intl";
import { act } from 'react-dom/test-utils';

import Box from '@material-ui/core/Box';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import IconButton from '@material-ui/core/IconButton';
import Button from '@material-ui/core/Button';
import Badge from '@material-ui/core/Badge';
import MenuItem from '@material-ui/core/MenuItem';
import Menu from '@material-ui/core/Menu';
import MenuIcon from '@material-ui/icons/Menu';
import NotificationsIcon from '@material-ui/icons/Notifications';
import MoreIcon from '@material-ui/icons/MoreVert';
import AccountCircle from '@material-ui/icons/AccountCircle';
import LanguageIcon from '@material-ui/icons/Language';

import ConnectedPublicHeader, { PublicHeader } from '../PublicHeader';
import { translationMessages } from 'config/i18n.config';
import theme from 'config/theme.config';
import { appLocales } from 'config/i18n.config';

// store mock
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
import {
    GLOBAL_CHANGE_LOCALE,
    USER_LOGIN_SUCCESS,
    USER_GET_USER_SUCCESS,
    USER_GET_PROFILE_SUCCESS
} from 'store/constants';
jest.mock('service/member');

configure({ adapter: new Adapter() });

describe('containers/layout/PrivateHeader', () => {

    const mount = createMount();

    const store = mockStore({
        user: undefined,
        global: { locale: 'en' }
    });

    const wrapper = mount(
        <Provider store={store}>
            <MemoryRouter>
                <IntlProvider locale='en' messages={translationMessages.en}>
                    <ThemeProvider theme={theme}>
                        {<ConnectedPublicHeader />}
                    </ThemeProvider>
                </IntlProvider>
            </MemoryRouter>
        </Provider>
    );

    test('snapshot test', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('go home', done => {
        act(() => {
            wrapper.find(PublicHeader).find(Toolbar).find('img').first().simulate('click');
        });

        setImmediate(() => {
            wrapper.update();
            done();
        });
    });

    test('go to map', done => {
        act(() => {
            wrapper.find(PublicHeader).find(Toolbar).find(Box).first().find(Button).last().simulate('click');
        });

        setImmediate(() => {
            wrapper.update();
            expect(wrapper.find('#language-menu').find(Menu).props().anchorEl).toBeFalsy();
            done();
        });
    });

    test('go to login', done => {
        act(() => {
            wrapper.find(PublicHeader).find(Toolbar).find(Box).at(1).find(Button).last().simulate('click');
        });

        setImmediate(() => {
            wrapper.update();
            expect(wrapper.find('#language-menu').find(Menu).props().anchorEl).toBeFalsy();
            done();
        });
    });

    test('locale menu - 1', done => {
        act(() => {
            wrapper.find(PublicHeader).find(Toolbar).find(Box).at(1).find(IconButton).last().simulate('click');
        });

        setImmediate(() => {
            wrapper.update();
            expect(wrapper.find('#language-menu').find(Menu).props().anchorEl).toBeTruthy();
            done();
        })
    });

    test('locale menu - select', done => {
        act(() => {
            wrapper.find('#language-menu').first().find(MenuItem).at(1).simulate('click');
        });

        setImmediate(() => {
            wrapper.update();
            expect(wrapper.find('#language-menu').find(Menu).props().anchorEl).toBeFalsy();
            expect(store.getActions()).toContainEqual({
                type: GLOBAL_CHANGE_LOCALE,
                payload: 'ko'
            });
            done();
        })
    });

    test('locale menu - 2', done => {
        act(() => {
            wrapper.find(PublicHeader).find(Toolbar).find(Box).at(1).find(IconButton).last().simulate('click');
        });

        setImmediate(() => {
            wrapper.update();
            expect(wrapper.find('#language-menu').find(Menu).props().anchorEl).toBeTruthy();
            done();
        })
    });

    test('locale menu - close by escape', done => {
        act(() => {
            wrapper.find('#language-menu').find(Menu).first().props().onClose({}, 'escapeKeyDown');
        });

        setImmediate(() => {
            wrapper.update();
            expect(wrapper.find('#language-menu').find(Menu).props().anchorEl).toBeFalsy();
            done();
        })
    });
});