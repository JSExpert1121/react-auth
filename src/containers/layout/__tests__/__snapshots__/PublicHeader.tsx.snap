// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/layout/PrivateHeader snapshot test 1`] = `ReactWrapper {}`;
